### PR TITLE
#57: deny old revocations

### PR DESCRIPTION
**Pull request by danieldelhoyo**
- Deny revocation if timestap < now - maxTimeDelta
- Created unit test

### CODE REVIEW

File: src/main/java/org/zalando/planb/revocation/api/exception/AncientRevocationException.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+/**
+ * Thrown when a client tries to post a revocation with an {@code issued_before} timestamp in the ancient past.
+ *
+ */
+public class AncientRevocationException extends RuntimeException {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final String MESSAGE = "'issued_before' cannot be set in the ancient past.";
```
Comment (gargravarr) [inline src/main/java/org/zalando/planb/revocation/api/exception/AncientRevocationException.java]: Didn't like the phrasing. Also I would put more context (If we have it). Would write:

> 'issued_before' too old. Value must be greater than <timestamp here>

Where <timestamp here> is now - maxTimeDelta

File: src/test/java/org/zalando/planb/revocation/api/RevocationResourceTest.java [hunk 2]
```diff
@@ -181,6 +182,26 @@ public void testBadRequestWhenPostingFutureRevocation() throws Exception {
     }
 
     /**
+     * Tests that when {@code POST}ing revocations with an ancient {@code issued_before} field, a HTTP {@code BAD_REQUEST}
+     * is returned.
+     *
+     * <p>Furthermore asserts that a standard {@link Problem} is returned.</p>
+     */
+    @Test
+    public void testBadRequestWhenPostingAncientRevocation() throws Exception {
```
Comment (gargravarr) [inline src/test/java/org/zalando/planb/revocation/api/RevocationResourceTest.java]: If maxTimeDelta > ONE_YEAR_IN_SECONDS this fails. So this must be relative to it.

**Comment by gargravarr** [comment]
:+1:

**Comment by vroldanbet** [comment]
Addresses https://github.com/zalando/planb-revocation/issues/57

**Pull request merged**